akka-stream/src/main/scala/akka/stream/SubscriptionWithCancelException.scala

```diff
@@ -19,6 +20,12 @@ trait SubscriptionWithCancelException extends Subscription {
   def cancel(cause: Throwable): Unit
 }
 object SubscriptionWithCancelException {
-  case object NoMoreElementsNeeded extends RuntimeException with NoStackTrace
-  case object StageWasCompleted extends RuntimeException with NoStackTrace
+
+  /**
+   * Not for user extension
+   */
+  @DoNotInherit
```

Comment:
Do we always add that to sealed hierarchies because of Java users?

Reply:
Yep